docs/DeveloperGuide.md

```diff
@@ -1,34 +1,511 @@
+
+Figure 11. Class Diagram of the ConfigStorageManager
+
+The Config Storage,
+* Checks existence of the config data file in the `data` directory. If the file does not exist, create a new data file for config storage.
+* Invokes the `configEncoder` class to encode the String of user name, the integer number of recommended hours and finally, the boolean which checks if the program has run before. The encoded string is written into a config data file.
+* Invokes the `configDecoder` class to decode the String user name, the integer number of recommended hours and the boolean which checks if the program has run before from a config data file. The three variables are used to show a different welcome message.
+
+
+The Config Parameter,
+* Helps to store the String of user name, the integer number of recommended hours and finally, the boolean which checks if the program has run before.
+* Contents stored in the parameter are subsequently used by configEncoder to encode them into a string. 
+
+
+---
+
+## Implementation
+
+This section describes some noteworthy details on how certain features are implemented.
+
+### User Interface
+The user interface of Plan&Score uses the singleton design. There is only one instance of UserInterface to be used throughout the application. The API for the user interface is UserInterface.java.
+
+#### Printing to user
+Instead of using the default method `System.out.println()` provided by Java to display messages to the user, use `showToUser(String …)` where multiple strings can be added into the arguments. Each string provided that is separated by `,` will be printed on the next line.
+
+Example input:
+```
+UserInterface userInterface = UserInterface.getInstance();
+
+String string1 = "Hello";
+String string2 = "How are you?";
+
+userInterface.showToUser(string1, string2);
+```
+
+Example output:
+```
+Hello
+How are you?
+```
+
+Code:
+```
+public void showToUser(String... message) {
+   for (String m : message) {
+       out.println(m);
+   }
+}
+```
+
+
+#### Printing arrays to user
+
+The method `printArray(ArrayList<String> stringArrayList)` is provided for the printing of any arraylists.
+
+Example input:
+```
+ArrayList<String> stringArrayList = new ArrayList<>();
+
+// Adding items into the stringArrayList
+stringArrayList.add("Hello");
+stringArrayList.add("How are you?);
+
+// Obtaining user interface instance
+UserInterface userInterface = UserInterface.getInstance();
+userInterface.printArray(stringArrayList);
+```
+
+Example output:
+```
+Hello
+How are you?
+```
+
+Code:
+```
+public void printArray(ArrayList<String> stringArrayList) {
+   assert stringArrayList != null;
+   for (String line : stringArrayList) {
+       userInterface.showToUser(line);
+   }
+}
+```
+
+### Help feature
+
+![helpcomponent](https://raw.githubusercontent.com/AY2021S1-CS2113T-W12-4/tp/master/docs/diagram/Help.png)
```

Comment:
![image](https://user-images.githubusercontent.com/60251547/97668800-bb648400-1abd-11eb-943b-689dd016a063.png)

should there be a return arrow (the part i circled) ?
and for showToUser(), should it be inside the handleHelp()?